.circleci/config.yml

```diff
@@ -22,293 +19,302 @@ parameters:
   key_vault_name:
     type: string
     default: "kv-circleci"
+  jmeter_directory_input:
+    type: string
+    default: "jmeter"
+  junit_file_name:
+    type: string
+    default: "output.xml"
+  properties_file_name:
+    type: string
+    default: "sample.properties"
+  environment_name:
+    type: string
+    default: "ne"
 
 orbs:
-  windows: << pipeline.parameters.windows_image >>
+  windows: circleci/windows@2.2.0
+  terraform: circleci/terraform@3.0.0
+  azure-cli: circleci/azure-cli@1.2.0
 
-jobs:
-  build:
-    description: Build application with Release configuration
-    executor:
-      name: windows/default
+commands:
+  login-to-azure:
     steps:
-      - checkout
+      - azure-cli/install
 
       - run:
-          name: "Build Application according to some given configuration"
-          command: >
-            dotnet
-            build << pipeline.parameters.project_dir >>
-            --configuration Release
-
-  test:
-    description: Setup and run application tests
-    executor:
-      name: windows/default
+          name: "Verify Azure install"
+          command: az -v
+
+      - azure-cli/login-with-service-principal:
+          azure-sp: ARM_CLIENT_ID
+          azure-sp-password: ARM_CLIENT_SECRET
+          azure-sp-tenant: ARM_TENANT_ID
+  
+  login-to-azure-down:
     steps:
-      - checkout
+      - azure-cli/install
+      - run: sudo apt-cache policy azure-cli | sudo apt-get install --allow-downgrades azure-cli=2.18.0-1~stretch
```

Comment:
Why do we need to downgrade az-cli?

Reply:
There's a bug in the latest version of az-cli that is causing an error when creating the Bastion host.

Reply:
https://github.com/Azure/azure-cli/pull/20305